ctapipe/image/extractor.py

```diff
@@ -524,6 +597,66 @@ def __call__(self, waveforms, telid, selected_gain_channel):
         return charge, peak_time
 
 
+class SlidingWindowMaxSum(ImageExtractor):
+    """
+    Sliding window extractor that maximizes the signal in window_width consecutive slices.
+    """
+
+    window_width = IntTelescopeParameter(
+        default_value=7, help="Define the width of the integration window"
+    ).tag(config=True)
+
+    apply_integration_correction = BoolTelescopeParameter(
+        default_value=True, help="Apply the integration window correction"
+    ).tag(config=True)
+
+    @lru_cache(maxsize=128)
+    def _calculate_correction(self, telid):
+        """
+        Calculate the correction for the extracted charge such that the value
+        returned would equal 1 for a noise-less unit pulse.
+
+        This method is decorated with @lru_cache to ensure it is only
+        calculated once per telescope.
+
+        WARNING: TO BE DONE properly, the current code reuses the function of
```

Comment:
Could you implement this directly here, does not sound to complicated?

Reply:
the main reason why I did not do so is because this feature does not seem to be used (at least in LST), so I did not have a proper set-up to test it, but I can look into making some dummy pulse shape and testing on it.

Reply:
that would be great

Reply:
You have the reference pulse shape in the `CameraDescription`. I guess it's a fairly small effect though, and the correction doesn't really matter much except to get the cleaning thresholds in the same units for all cameras.

Reply:
Sorry for the delay, I restarted working on this. 
CameraDescription.readout is where the code is taking the pulse shape from. However this is not really reliable. 
If I execute the following code 
`import numpy as np
import astropy.units as u
import matplotlib.pyplot as plt
plt.ion()

from ctapipe.instrument import SubarrayDescription, TelescopeDescription

subarray = SubarrayDescription(
        "LST1",
        tel_positions={1: np.zeros(3) * u.m},
        tel_descriptions={
            1: TelescopeDescription.from_name(
                optics_name="LST", camera_name="LSTCam"
            ),
        },
)
pulse_shape=subarray.tel[1].camera.readout.reference_pulse_shape[ dt=subarray.tel[1].camera.readout.reference_pulse_sample_width
xs=np.arange(len(pulse_shape))*dt
plt.plot(xs, pulse_shape)`
I get the following figure:
![pulse_shape_LST](https://user-images.githubusercontent.com/33022433/106497474-5fdfd300-64be-11eb-854f-c24ac1e1325a.png)
which is a much broader pulse then it should 

The calculation of the correction factor would be much simpler if the pulse shape in this class had the same binning as the actual readout, this is the case in the above example, and one would assume to take it from granted since the shape is taken from the "readout" object, which has the binning embedded, however in the first tests that i was doing in lstchain, when the array was being read from the data the pulse shapes there were actually a delta function with a SSC-like sampling, so obviously  it cannot be taken for granted. 

I will change the code to use a simple conversion and rounding of sampling to make it work also in this more general case, but the whole issue of the LST pulse shape deserves a separate "issue"

Reply:
The `from_name()` methods load up a file from ctapipe-extra (which is now a directory on a server rather than a package), and are just meant for unit-testing purposes.  Currently everything in ctapipe-extra is from PROD3 or even PROD2 simulations, so quite out of date for real analysis.  In the future I want to clean that up and have an option to select which "prod" to use, but there has not been manpower for that (see e.g #738 )

If you load real data from a SimTel file or something else supported, the correct waveform should be loaded into the instrument model that you get from `source.subarray`.

Reply:
Note taht if you don't want to always load up a file, you can take any reference prod5 file, for example, and run 
```sh
ctapipe-dump-instrument --input=some_reference_file.simtel.gz
```
And it will dump a bunch of FITS files including the Camera geometry and readout definitions to the local directory.  You can then setenv `CTAPIPE_SVC_PATH=[directory where those files are]`, and ctapipe will use that when you run the `from_name()` functions instead of the defaults.

Reply:
One more thing: be careful how you plot it - the reference pulse shape in the MCs are *oversampled*, so **not** at the same sampling rate as the waveform! I can't tell in your plot where you get dt from, but I suspect you are using the dt from the waveform, not from the reference pulse.

Reply:
Update: I found the actual problem... In fact, the camera readout definition for LSTCam does not even exist in the ctapipe-extra directory on the dataserver. It seems the default behavior is to just return some dummy pulse shape, which has no real meaning (you should see a logger warning message if logging is set up)...   Clearly this is not good behavior. 

here is a comparison with prod3b:
![image](https://user-images.githubusercontent.com/11677812/106580313-0d47fa80-6542-11eb-914e-a2f0f973891c.png)

Reply:
This is also why you get only 1 reference pulse shape if you use `from_name()`... 
I had already opened an issue about this, but obviously forgot: See #1450